File: lab1/php/InputValidator.php
<?php
class InputValidator {
    private string $delta_x;
    private string $delta_y;
    private string $delta_r;

    public function __construct(string $delta_x, string $delta_y, string $delta_r)
    {
        $this->delta_x = $delta_x;
        $this->delta_y = $delta_y;
        $this->delta_r = $delta_r;
    }

    public function is_valid(): bool {
        return $this->check_x() && $this->check_y() && $this->check_r();
    }

    private function check_x(): bool {
        return is_numeric($this->delta_x)
            && in_array($this->delta_x, array(-2, -1.5, -1, -0.5, 0, 0.5, 1, 1.5, 2));
    }

    private function check_y(): bool {
        return is_numeric($this->delta_y)
            && $this->delta_y >= -5 && $this->delta_y <= 3;
    }

    private function check_r(): bool {
        return is_numeric($this->delta_r)
            && in_array($this->delta_r, array(1, 1.5, 2, 2.5, 3));
    }
}

File: src/main/java/utils/InputValidator.java
package utils;

import java.math.BigDecimal;
import java.util.Arrays;
import java.util.List;

public class InputValidator {
    private final BigDecimal x;
    private final BigDecimal y;
    private final BigDecimal r;

    public InputValidator(BigDecimal x, BigDecimal y, BigDecimal r) {
        this.x = x;
        this.y = y;
        this.r = r;
    }

    public boolean isValid() {
        return checkX() && checkY() && checkR();
    }

    private boolean checkX() {
        List<BigDecimal> validXValues = Arrays.asList(
                new BigDecimal("-2.0"), new BigDecimal("-1.5"), new BigDecimal("-1.0"),
                new BigDecimal("-0.5"), new BigDecimal("0.0"), new BigDecimal("0.5"),
                new BigDecimal("1.0"), new BigDecimal("1.5"), new BigDecimal("2.0")
        );
        for (BigDecimal validX : validXValues) {
            if (validX.compareTo(this.x) == 0) {
                return true;
            }
        }
        return false;
    }

    private boolean checkY() {
        BigDecimal minY = new BigDecimal("-5.0");
        BigDecimal maxY = new BigDecimal("3.0");
        return this.y.compareTo(minY) >= 0 && this.y.compareTo(maxY) <= 0;
    }

    private boolean checkR() {
        List<BigDecimal> validRValues = Arrays.asList(
                new BigDecimal("1.0"), new BigDecimal("2.0"), new BigDecimal("3.0"),
                new BigDecimal("4.0"), new BigDecimal("5.0")
        );
        for (BigDecimal validR : validRValues) {
            if (validR.compareTo(this.r) == 0) {
                return true;
            }
        }
        return false;
    }
}
